[MRG] Start glossary (#391)
jonescompneurolab/hnn-core: doc/glossary.rst

Glossary
========

.. currentmodule:: hnn_core

The Glossary provides short definitions of HNN-specific vocabulary and
general computational neuroscience concepts. If you think a term is missing,
please consider creating a new issue or opening a pull request to add it.

.. glossary::
    :sorted:

    drive
        An artificial cell or population of artificial cells that only produces
        spikes according to a given pattern and is used to initiate and sustain
        cell spiking in the network

    gid
        The "global" cell ID. Global refers to the fact that the cell ID
        is unique even if the simulation is split across cores.

    distal

>         anatomical location that is further away from the soma

Reviewer: ```suggestion
        Anatomical location that is further away from the soma.
        Distal drives refer to extrinsic connections to the distal dendrites of
        the cortical column, which often originate from direct thalamic nuclei.
```

    proximal

>         anatomical location that is closer to the soma

Reviewer: ```suggestion
        Anatomical location that is closer to the soma.
        Proximal drives refer to extrinsic connections to the proximal dendrites
        of the cortical column, which often originate from indirect thalamic
        nuclei, as well as other cortical regions.
```

Glossary
========

.. currentmodule:: hnn_core

The Glossary provides short definitions of HNN-specific vocabulary and
general computational neuroscience concepts. If you think a term is missing,
please consider creating a new issue or opening a pull request to add it.

.. glossary::
    :sorted:

    drive
        An artificial cell or population of artificial cells that only produces
        spikes according to a given pattern and is used to initiate and sustain

>         cell spiking in the network

Reviewer 2: Is "drive" an artificial cell or the synaptic spikes that emerge from an artificial cell? This is nuanced, but I think of "drive" as modeling a physiological process rather than a physiological structure.

Author: agreed, and we may even want to avoid saying "artificial cells" ... it's bringing another new term to explain it. Can you make a suggested edit?

Reviewer 2: ```suggestion
    drive
        An exogenous source of excitatory action potentials (spikes)
        that synapse onto neuronal compartments, perturbing the network.
```

Author: okay I deleted the second sentence to keep it simple for now. Thanks for the suggestion!

    gid

>         The "global" cell ID. Global refers to the fact that the cell ID
>         is unique even if the simulation is split across cores.

Reviewer 2: ```suggestion
        The "global" cell ID. A gid is assigned to each cell in the
        network as well as artificial drive cells. Global refers to
        the fact that the cell ID is unique even if the simulation
        is split across cores.
```

    distal
        Anatomical location that is further away from the soma.
        Distal drives refer to extrinsic connections to the distal dendrites of

>         the cortical column, which often originate from direct thalamic nuclei.

Reviewer 2: ```suggestion
        the cortical column, which often originate from indirect thalamic nuclei.
```

    proximal
        Anatomical location that is closer to the soma.
        Proximal drives refer to extrinsic connections to the proximal dendrites

>         of the cortical column, which often originate from indirect thalamic
>         nuclei, as well as other cortical regions.

Reviewer 2: ```suggestion
        of the cortical column, which often originate from direct thalamic
        nuclei.
```

Author: is "direct"/"indirect" an anatomical location? I often get confused by this kind of phrasing that seems quite common on neuroscience ... you google "direct thalamic nuclei" and you get "direct pathway" and "thalamic nuclei" but no "direct thalamic nuclei" ...

Reviewer 2: > is "direct"/"indirect" an anatomical location? I often get confused by this kind of phrasing that seems quite common on neuroscience ... you google "direct thalamic nuclei" and you get "direct pathway" and "thalamic nuclei" but no "direct thalamic nuclei" ...

I'm not convinced that "direct" and "indirect" are the proper terms to use here. Maybe we should stick with "lemniscal" and "non-lemniscal" as described in the HNN-GUI tutorials.

This distinction does not separate two anatomical regions, rather, it's an informal classification scheme that some anatomists use to categorize thalamic nuclei based on their afferent projections into cortex. For instance, the ventral posteromedial nucleus (VPM) is part of the lemniscal pathway that has bulk projections to the granular layer of S1. Conversely, the posterior medial nucleus (POm) has dispersed projections (that are not part of the lemniscal pathway) to numerous cortical regions that happen to include the superficial layers of S1. This might be a good place for @stephanie-r-jones to weigh in.
